### ISSUE TITLE
automatic refresh using PowerBI service

### ISSUE THREAD

**Issue by ViValentin** [MattWade-MS/QER#2]
Hello,
Version 4.3 introduce ASN to ISP translation but it seems this functionality doesn't support the use of PowerBI service instead of PowerBI desktop as it use dynamic data sources.
As consequence we can't publish QER to a users who don't have authorization to access data.
Is there a way to correct this?
Regards

**Comment by MattWade-MS** [comment]
I have successfully published the current version of the report to PBI Online as is.  However the DBIP database will not auto-refresh every month, what you get is a warning, not an error or a full stop.  To get around this you can just re-publish once a month by opening the report up in PBI desktop first (so the refresh works) and then re-publish.

Additionally, you could edit the DBIP query properties.  Comment out the first three rows and modify the 4th row as follows.  This will disable the auto-refresh of the DBIP and make it static.  You'll have to change the year-month value every month.  So either way you are going to have to change something monthly.  I found the default republish option the easiest to manage.

Before:
![image](https://user-images.githubusercontent.com/34693246/119183251-cf63d900-ba41-11eb-989c-eb41fb2cb098.png)

After:
![image](https://user-images.githubusercontent.com/34693246/119183289-e0144f00-ba41-11eb-930e-0d69a83ab391.png)

**Comment by ViValentin** [comment]
Hello MatWade,
Thanks for your explanations. Based on them I founded out a third alternative.

I written a small Powershell script downloading the compressed CSV file and publishing it to a Teams SharePoint library using SharepointPNP Powershell module so that it doesn't need any PowerBI Gateway for the update. Then I changed the source and schedule my Powershell  script.

As the file within the SharePoint library have always the same name I don't need to update anymore the PBIX. I will just need to change your template when you will do a never version.

Here is the few line for the downloading part:
$TodayYearMonth  = (Get-Date).ToString('yyyy-MM')
$Source = 'https://download.db-ip.com/free/dbip-asn-lite-'+$TodayYearMonth+'.csv.gz'
$destination = 'C:\Temp\dbip-asn-lite.csv.gz'
Invoke-RestMethod -Uri $source -OutFile $destination
Regards
